src/main/java/org/prebid/server/bidder/adf/AdfBidder.java

```diff
@@ -49,17 +53,30 @@ public AdfBidder(String endpointUrl, JacksonMapper mapper) {
     public Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
         final List<Imp> modifiedImps = new ArrayList<>();
         final List<BidderError> errors = new ArrayList<>();
+        String priceType = "";
 
         for (Imp imp : bidRequest.getImp()) {
             try {
                 final ExtImpAdf adfImp = parseExt(imp);
                 final Imp modifiedImp = imp.toBuilder().tagid(adfImp.getMid()).build();
                 modifiedImps.add(modifiedImp);
+                priceType = StringUtils.isEmpty(priceType)
+                        && Objects.nonNull(adfImp.getPriceType()) ? adfImp.getPriceType() : "";
             } catch (PreBidException e) {
                 errors.add(BidderError.badInput(e.getMessage()));
             }
         }
 
+        if (StringUtils.isNotEmpty(priceType)) {
+            final ObjectNode adfNode = mapper.mapper().createObjectNode()
+                    .set("pt", TextNode.valueOf(priceType));
+
+            final ExtRequest fillExtRequest =
+                    mapper.fillExtension(ExtRequest.empty(), adfNode);
+
+            bidRequest = bidRequest.toBuilder().ext(fillExtRequest).build();
```

Comment:
Introduce new variable `modifiedRequest`. Do not assign new reference to method parameters.